### PR TITLE
7 allow to index incrementally

### LINKED ISSUE #7: Allow to index incrementally
Allow to index incrementally. Now it is only possible to index and then replace the old index.
Increment should not change data until it was successful.

### PR DESCRIPTION
**Pull request by mighantos**
Adds feature of indexing incrementally. (until now it was only possible to override existing index)

Closes #7

### CODE REVIEW

File: src/main/java/org/elasticsearch/app/api/server/exceptions/CouldNotSearchForIndex.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    public CouldNotSearchForIndex(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public CouldNotSearchForIndex(Throwable cause) {
+        super(cause);
+    }
+
+    public CouldNotSearchForIndex(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
```
Comment (blcham) [inline src/main/java/org/elasticsearch/app/api/server/exceptions/CouldNotSearchForIndex.java:23]: All of the exceptions have this method with 4 parameters but you never use it in the code, so what is the purpose of duplicating all constructors of RuntimeException?

File: src/main/java/org/elasticsearch/app/api/server/exceptions/CouldNotSetSettingsOfIndex.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    public CouldNotSetSettingsOfIndex(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public CouldNotSetSettingsOfIndex(Throwable cause) {
+        super(cause);
+    }
+
+    public CouldNotSetSettingsOfIndex(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
```
Comment (blcham) [inline src/main/java/org/elasticsearch/app/api/server/exceptions/CouldNotSetSettingsOfIndex.java:23]: see comment above

File: src/main/java/org/elasticsearch/app/Harvester.java [hunk 3]
```diff
@@ -562,6 +559,11 @@ public Harvester rdfIndexType(String indexType) {
         return this;
     }
 
+    public Harvester setIncrementally(boolean incrementally) {
```
Comment (blcham) [inline src/main/java/org/elasticsearch/app/Harvester.java:562]: I saw quite many very similar functions here, repeating same pattern:
rdfNormalizationObj
rdfNormalizationMissing
rdfNormalizationProp

I would think of [Lombok](https://projectlombok.org/features/GetterSetter) to do this ... and set up getters with props like:
lombok.accessors.chain
lombok.accessors.fluent

**Review by blcham — APPROVED** [review]
LGTM

Do not understand in necessary detail the content of the commit, so I commented rather "code style".  The issues mentioned are not important at all.

File: src/main/java/org/elasticsearch/app/Harvester.java (hunk 3 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/Harvester.java:562]: Other similar methods are from someone else from the forked repo so I used the same pattern. I am planning a big refactor so I will add this to the list.

File: src/main/java/org/elasticsearch/app/api/server/exceptions/CouldNotSearchForIndex.java (hunk 1 above)
Comment (mighantos) [reply src/main/java/org/elasticsearch/app/api/server/exceptions/CouldNotSearchForIndex.java:23]: Noted

**Pull request merged**